db.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+import json
+import sys
+
+
+def validate(input):
+    normalized = normalize_internal(input)
+    if normalized != input:
+        sys.stderr.write("Input is not normalized.\n")
+        sys.exit(0)
+
+def normalize_internal(input):
+    db = json.loads(input)
+    string = json.dumps(normalize_db(db), ensure_ascii=False, allow_nan=False, sort_keys=False, indent=2)
+    return string + "\n"
+
+def normalize(input):
+    open("db.json", "w").write(normalize_internal(input))
+
+def normalize_db(db):
+    i = 0
+    for workstream in db["workstreams"]:
```

Comment:
This is pretty verbose, and looking at it I wonder if there's something special going on but I think not. How about just `db["workstreams"] = [normalize_workstream(w) for w in db["workstreams"]]` or `db["workstreams"] = list(map(normalize_workstream , db["workstreams"]))`?